testsuite/manualmode/src/test/java/org/wildfly/core/test/standalone/mgmt/ManagementMultipartRequestTestCase.java

```diff
@@ -0,0 +1,410 @@
+        builder.addPart("uploadFormElement", createDeploymentContentPart());
+        builder.addPart("operation", createDeploymentOperationPart());
+        HttpEntity entity = builder.build();
+        HttpPost post = new HttpPost(managementUri);
+        post.setHeader("X-Management-Client-Name", "HAL");
+        post.setEntity(entity);
+
+
+        // client intentionally doesn't use certificate to authenticate, even though truststore auth is configured,
+        // but uses DIGEST instead
+
+        HttpClient client = SSLTruststoreUtil.getHttpClientWithSSL(CLIENT_TRUSTSTORE_FILE.getAbsoluteFile(),
+                SecurityTestConstants.KEYSTORE_PASSWORD);
+
+        // credentials
+        CredentialsProvider credsProvider = new BasicCredentialsProvider();
+        credsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(USERNAME, PASSWORD));
+
+        // digest auth
+        AuthCache authCache = new BasicAuthCache();
+        DigestScheme digestScheme = new DigestScheme();
+        authCache.put(new HttpHost(managementUri.getHost()), digestScheme);
+
+        final HttpClientContext context = HttpClientContext.create();
+        context.setCredentialsProvider(credsProvider);
+        context.setAuthCache(authCache);
+
+        // request
+        try {
+            HttpResponse response = client.execute(post, context);
+            Assert.assertEquals(200, response.getStatusLine().getStatusCode());
+        } finally {
+            ((CloseableHttpClient) client).close();
+        }
+    }
+
+    private static ContentBody createDeploymentOperationPart() throws IOException {
+        ModelNode op = Util.createAddOperation(PathAddress.pathAddress(DEPLOYMENT, DEPLOYMENT_NAME));
+        op.get("enabled").set(true);
+        op.get("content").add().get(INPUT_STREAM_INDEX).set(0);
+
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        op.writeBase64(baos);
+
+        return new ByteArrayBody(baos.toByteArray(), ContentType.create("application/dmr-encoded"), "blob");
+    }
+
+    private static ContentBody createDeploymentContentPart() {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+        WebArchive war = createDeployment();
+        war.as(ZipExporter.class).exportTo(baos);
+        LOG.infof("Deployment size is %d bytes", baos.size());
+        Assert.assertTrue("Generated war is too small, add some more classes!", baos.size() > 20000);
+
+        return new ByteArrayBody(baos.toByteArray(), ContentType.create("application/x-webarchive"), "deployment.war");
+    }
+
+    private static WebArchive createDeployment() {
+        // include whatever classes, but need to reach certain minimal size
+        return ShrinkWrap.create(WebArchive.class, DEPLOYMENT_NAME)
+                .addPackage(ShrinkWrap.class.getPackage());
+    }
+
+    private static void keyMaterialSetup() throws Exception {
+        FileUtils.deleteDirectory(WORK_DIR);
```

Comment:
the WORK_DIR should also be cleaned up when the test is tear down

Reply:
Done.